tests/lib/utils/casing.js

```diff
@@ -30,30 +34,56 @@ describe('getConverter()', () => {
     assert.equal(converter('FooBAR'), 'FooBAR')
     assert.equal(converter('Foo1BAZ'), 'Foo1BAZ')
     assert.equal(converter('foo1b_a_z'), 'Foo1bAZ')
+    assert.equal(converter('darИībaÊÊw'), 'DarИībaÊÊw')
+    assert.equal(converter('    foo       Bar   '), 'FooBar')
   })
 
   it('should convert string to kebab-case', () => {
     const converter = casing.getConverter('kebab-case')
 
+    assert.equal(converter('foo'), 'foo')
     assert.equal(converter('fooBar'), 'foo-bar')
     assert.equal(converter('foo-bar'), 'foo-bar')
     assert.equal(converter('foo_bar'), 'foo-bar')
     assert.equal(converter('FooBar'), 'foo-bar')
-    assert.equal(converter('Foo1Bar'), 'foo1bar')
+    assert.equal(converter('Foo1Bar'), 'foo1-bar')
     assert.equal(converter('FooBAR'), 'foo-b-a-r')
-    assert.equal(converter('Foo1BAZ'), 'foo1b-a-z')
+    assert.equal(converter('Foo1BAZ'), 'foo1-b-a-z')
     assert.equal(converter('foo1b_a_z'), 'foo1b-a-z')
+    assert.equal(converter('darИībaÊÊw'), 'dar-иība-ê-êw')
+    assert.equal(converter('    foo       Bar   '), 'foo-bar')
   })
 
   it('should convert string to snake_case', () => {
     const converter = casing.getConverter('snake_case')
 
+    assert.equal(converter('a'), 'a')
     assert.equal(converter('fooBar'), 'foo_bar')
     assert.equal(converter('foo-bar'), 'foo_bar')
     assert.equal(converter('FooBar'), 'foo_bar')
-    assert.equal(converter('Foo1Bar'), 'foo1bar')
+    assert.equal(converter('Foo1Bar'), 'foo1_bar')
     assert.equal(converter('FooBAR'), 'foo_b_a_r')
-    assert.equal(converter('Foo1BAZ'), 'foo1b_a_z')
+    assert.equal(converter('Foo1BAZ'), 'foo1_b_a_z')
     assert.equal(converter('foo1b_a_z'), 'foo1b_a_z')
+    assert.equal(converter('darИībaÊÊw'), 'dar_иība_ê_êw')
+    assert.equal(converter('    foo       Bar   '), 'foo_bar')
+  })
+
+  it('unicode tests', () => {
+    const words = [
+      ['ÊtreSîne', 'être_sîne'],
+      ['darbībaÊÊw', 'darbība_ê_êw'],
```

Comment:
This is correct with a popular casing. However, I think there are times when there is a difference with the solution method of Vue.
https://github.com/vuejs/vue/blob/master/src/shared/util.js#L149

I'm worried about being broken by some autofix.

Reply:
hmm, how it should work than xd

Reply:
vue is not checking for utf characters, they will have to polyfill.

Reply:
@ota-meshi maybe we should just ignore non a-z utf characters?

Reply:
It's difficult to decide.
@michalsnik what do you think?

Reply:
In most of Japan, it is forbidden to use a source other than a-z to avoid problems.

Reply:
@armano2 
> what do you think about making new rule which forbids usage of non ascii characters in code?

I think that rule is very nice!

Reply:
I think that we should follow the original compiler's behavior.

> what do you think about making new rule which forbids usage of non ascii characters in code?

Sounds good to me.

Reply:
Perhaps if we introduce this rule and put it in the `base` category - we wouldn't need this fix at all?